File: dbf/clients.py
from dbfread import DBF
import pandas as pd
import mysql.connector

dbConfig = {
    "host": "localhost",
    "user": "root",
    "password": "supersecretpassword",
    "database": "dashboard",
    "charset": "utf8mb4"
}

file = './dbf/DBF_VOLLVILLESPRL/TIE.DBF'
table = DBF(file, encoding='latin-1')

conn = mysql.connector.connect(**dbConfig)
cursor = conn.cursor()

sql = """
INSERT INTO clients (
    laboruId,
    name,
    description,
    email,
    vat,
    vatType,
    vatRate,
    phone1,
    phone2,
    phone3,
    phone4,
    lang,
    websiteUrl,
    createdAt,
    updatedAt
) VALUES (
    %(laboruId)s,
    %(name)s,
    %(description)s,
    %(email)s,
    %(vat)s,
    %(vatType)s,
    %(vatRate)s,
    %(phone1)s,
    %(phone2)s,
    %(phone3)s,
    %(phone4)s,
    %(lang)s,
    %(websiteUrl)s,
    NOW(),
    NOW()
)
"""

for record in table:
    vat = f"{record.get('TIEREFTVA','')}{record.get('TIENUMTVA','')}".strip()
    vatRate = "0%"
    if record.get("TVCCLE__S") and record.get("TVCCLE__S") != "":
        vatRate = record.get("TVCCLE__S")   

    data = {                
        "laboruId": record["TIECLE"],
        "name": record.get("TIENOM__1"),
        "description": record.get("TIENOM__2"),
        "email": record.get("WEBMEL"),
        "vat": vat or None,
        "vatType": record.get("TIEREGTVA"),
        "vatRate": vatRate,
        "phone1": record.get("TIENUMTE1"),
        "phone2": record.get("TIENUMTE2"),
        "phone3": record.get("TIENUMTE3"),
        "phone4": record.get("TIENUMTE4"),
        "lang": record.get("TIELNG").replace('FRA', 'fr'),
        "websiteUrl": record.get("WEBMEL"),
    }

    cursor.execute(sql, data)

conn.commit()
cursor.close()
conn.close()

File: front/models/clients.ts
import { Model, Collection } from "../helpers/models/models";
import { ModelProperty } from "@/helpers/models/model";

export class ClientModel extends Model {
  protected static url: string = "clients";
  protected static key: string = "client";
  protected static attributeId: string = "id";  

  constructor(id?: number) {
    super();

    this.addProperty(new ModelProperty<number | null>("id", id || -1)); 
    this.addProperty(new ModelProperty<string>("laboruId", ""));
    this.addProperty(new ModelProperty<string>("name", ""));
    this.addProperty(new ModelProperty<string>("email", ""));
    this.addProperty(new ModelProperty<string>("description", ""));
    this.addProperty(new ModelProperty<string>("vat", ""));
    this.addProperty(new ModelProperty<string>("vatType", ""));
    this.addProperty(new ModelProperty<string>("vatRate", ""));  
    this.addProperty(new ModelProperty<string>("phone1", "")); 
    this.addProperty(new ModelProperty<string>("phone2", "")); 
    this.addProperty(new ModelProperty<string>("phone3", "")); 
    this.addProperty(new ModelProperty<string>("phone4", "")); 
    this.addProperty(new ModelProperty<boolean>("archived", false)); 
    this.addProperty(new ModelProperty<string>("createdAt", "")); 
    this.addProperty(new ModelProperty<string>("archivedAt", "")); 
    this.addProperty(new ModelProperty<string>("updatedAt", "")); 
  }
}

export class ClientCollection extends Collection<ClientModel> {
  static url: string = "clients";

  loadModel(id: any): ClientModel {
    return new ClientModel(id);
  }
}